### PR TITLE
NIFI-1251 - Allow ExecuteSQL to send out large result sets in chunks

### PR DESCRIPTION
**Pull request by patricker**
### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by mattyb149** [comment]
What do you think about naming conventions consistent between the processors that do this "incremental commit"? I added this capability to QueryDatabaseTable under [NIFI-4836](https://issues.apache.org/jira/browse/NIFI-4836), and not to say we have to use my namings, but as it is in product it might be better to have the properties and such be named and documented consistently. For example, QDT has a single property that takes a zero value to indicate that it should wait until all results are processed before sending the files downstream. What are your thoughts?

**Comment by patricker** [comment]
@mattyb149 It's a different approach, but I see no issues. It's also more flexible, as it would add in the option for "Max Rows Per Flow File" in addition to output batching. I'll work on it.

**Comment by patricker** [comment]
@mattyb149 Updated

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExecuteSQL.java [hunk 1]
```diff
@@ -146,6 +162,31 @@
...
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    public static final PropertyDescriptor OUTPUT_BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("esql-output-batch-size")
+            .displayName("Output Batch Size")
+            .description("The number of output FlowFiles to queue before committing the process session. When set to zero, the session will be committed when all result set rows "
+                    + "have been processed and the output FlowFiles are ready for transfer to the downstream relationship. For large result sets, this can cause a large burst of FlowFiles "
+                    + "to be transferred at the end of processor execution. If this property is set, then when the specified number of FlowFiles are ready for transfer, then the session will "
+                    + "be committed, thus releasing the FlowFiles to the downstream relationship. NOTE: The maxvalue.* and fragment.count attributes will not be set on FlowFiles when this "
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExecuteSQL.java]: Minor issue, we don't write maxvalue.* attributes in ExecuteSQL, I'll remove that reference while merging

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestExecuteSQL.java [hunk 2]
```diff
@@ -17,6 +17,7 @@
 package org.apache.nifi.processors.standard;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestExecuteSQL.java]: Some unused imports here, I'll remove them on merge unless there is a test or something missing that would use them?

**Comment by mattyb149** [comment]
+1 LGTM, ran build with unit tests and contrib-check (after making some minor edits), also ran various scenarios with ExecuteSQL in a live NiFi instance. Thanks for this very helpful contribution! Merging to master

**Pull request closed without merging**